### DISCUSSION TITLE
Identity card authentication

### DISCUSSION THREAD

**Discussion by EuroNode** [PiranhaCMS/piranha.core#1617] [Q&A] [answered]
We have a requirement to use identity card for user authentication for CMS. We liked the Piranha CMS for it's simplicity and implemented the alternative user authentication as described in Securing Pages section of documentation. It works ok. However whatever we do the manager part does not recognize authenticated user and always redirects to password login form.

Please help us solve this issue because else we will have to abandon Piranha.

At the same time please note that SqlServer identity provider does not work out of the box due to missing maxLength attributes on LoginProvider and ProviderKey fields in UserLogins table and LoginProvider and Name fields in UserTokens table. When length limit is missing index creations fail due to key size limit.

**Comment by tidyui** [comment, marked as answer]
Hi there! The manager doesn’t care about authentication, it only cares about claims. Here are the claims used by, and needed by the manager.

https://piranhacms.org/docs/master/architecture/authentication

As far as migration goes we’ll take a look if it has changed since the migration was generated. It was a couple of version back

**Comment by EuroNode** [comment]
Thanks for your help. Finally found that call to App.Init(api); in startup got lost so all manager claims were not created in database. Now it works.

 

One more question: I see traces of code for global search in the last version. Is there any timeline to finish it? That is also a mandatory requirement for our project and we can't use Azure because this will be run within high security network.
